### ISSUE TITLE
How can i use multiple dockerfiles for different platforms?

### ISSUE THREAD

**Issue by SeanoNET** [crazy-max/ghaction-docker-buildx#96]
Hi @crazy-max 

Thanks for this awesome action!

This might be a buildx specific query but if you could offer any insight it will be greatly appreciated.

I have added multi-platform package support based on your repo [ddns-route53](https://github.com/crazy-max/ddns-route53/blob/master/.github/workflows/build.yml) using this action.

I noticed that your Dockerfile sets the platform before pulling the base image to compile/package your application. 

`FROM --platform=${BUILDPLATFORM:-linux/amd64} golang:1.13.5-alpine as builder`

My application is a ASP Net Core application running on dotnet core 3.1 - It appears that Microsoft offers their image platforms based on different tags [microsoft-dotnet-core-runtime](https://hub.docker.com/_/microsoft-dotnet-core-runtime/?tab=description).

Is there anyway i can specify multiple docker files for each platform in buildx using your action?

[Dockerfile.debian-arm32](https://github.com/SeanoNET/RoomTempDashboard/blob/master/Dockerfile.debian-arm32)
`FROM mcr.microsoft.com/dotnet/core/aspnet:3.1-buster-slim-arm32v7` 

[Dockerfile](https://github.com/SeanoNET/RoomTempDashboard/blob/master/Dockerfile)
`FROM mcr.microsoft.com/dotnet/core/aspnet:3.1` 

At the moment i am getting the error below due to docker not finding a target platform in [Microsofts dotnet core SDK image](https://hub.docker.com/_/microsoft-dotnet-core-sdk/).

`FROM mcr.microsoft.com/dotnet/core/sdk:3.1 AS build`

https://github.com/SeanoNET/RoomTempDashboard/runs/436004074?check_suite_focus=true

```
failed to solve: rpc error: code = Unknown desc = failed to solve with frontend dockerfile.v0: failed to build LLB: failed to load cache key: no match for platform in manifest sha256:f4798993581a660740f06bdc126cf8258788a67518fb43e4275ecc6d4b226211: not found
```

**Comment by crazy-max** [comment]
Hi @SeanoNET,

> Is there anyway i can specify multiple docker files for each platform in buildx using your action?

I don't think you can do it in the current state of buildx. You should ask on [docker/buildx](https://github.com/docker/buildx).

> At the moment i am getting the error below due to docker not finding a target platform in Microsofts dotnet core SDK image.

As I can see the image `mcr.microsoft.com/dotnet/core/sdk:3.1` only handles the following platforms:

```
λ docker run --rm mplatform/mquery mcr.microsoft.com/dotnet/core/sdk:3.1
Image: mcr.microsoft.com/dotnet/core/sdk:3.1
 * Manifest List: Yes
 * Supported platforms:
   - linux/amd64
   - linux/arm/v7
   - linux/arm64
   - windows/amd64:10.0.17763.973
   - windows/arm
   - windows/amd64:10.0.18362.592
   - windows/amd64:10.0.18363.592
```

So you can only use docker `linux/amd64,linux/arm/v7,linux/arm64` platforms for now.

> Thanks for this awesome action!

And thanks for your support!

**Comment by SeanoNET** [comment]
Thanks the pointers, i appreciate it.

I managed to get further along with the docker buildx command after adjusting the platforms to the output of `docker run --rm mplatform/mquery mcr.microsoft.com/dotnet/core/runtime:3.1` but ended up hitting a limitation with the dotnet cli that had something to do with building on arm inside of QEMU.

I have reverted to a multi-build and push using the docker cli. It's messy but works.

I have attached my workflow for reference [docker.yml](https://github.com/SeanoNET/RoomTempDashboard/blob/master/.github/workflows/docker.yml)

Thanks again!
